### PR TITLE
Issue#13 Open api parent module skeleton

### PR DESCRIPTION
**Pull request by Pavlo-Ponomarenko**
dev
## GitHub

* [GitHub issue](https://github.com/ITA-Dnipro/Java-Home-Project-CQRS-Javelin/issues/13)

## Summary of issue

Implementing open api parent module skeleton
## Summary of change

Open api module generation. Requests for news implementation

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: open-api/src/main/resources/yaml/models/common.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      type: string
+      example: "example"
+
+BaseReadView:
+  type: object
+  properties:
+    id:
+      type: integer
+      format: int64
+      example: 1
```
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/models/common.yaml]: Please, add empty final line. That's basic GitHub requirement.

File: open-api/src/main/resources/yaml/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+tags:
+  - name: news
+    description: News management operations
+
+paths:
+  /news:
+    $ref: './paths/news.yaml#/news'
+
+  /news/{id}:
+    $ref: './paths/news.yaml#/news-newsId'
```
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/openapi.yaml]: Please, add empty final line. That's basic GitHub requirement.

File: open-api/src/main/resources/yaml/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+openapi: 3.0.3
+
+info:
+  title: Java-Home-Project-CQRS-Javelin-reader
```
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/openapi.yaml:4]: This open-api inplementation will be common for few modules, so you no need to specify reader module here.

File: open-api/src/main/resources/yaml/parameters/parameters.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  name: text
+  schema:
+    type: string
+  description: The content of the News
+title:
+  in: query
+  name: title
+  schema:
+    type: string
+  description: The title of the News
```
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/parameters/parameters.yaml]: Please, add empty final line. That's basic GitHub requirement.

File: open-api/src/main/resources/yaml/responses/responses.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      schema:
+        $ref: '../models/news.yaml#/ReadNews'
+NewsListResponse:
+  description: OK
+  content:
+    application/json:
+      schema:
+        type: array
+        items:
+          $ref: '../models/news.yaml#/ReadNews'
```
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/responses/responses.yaml]: Please, add empty final line. That's basic GitHub requirement.

File: open-api/src/main/resources/yaml/openapi.yaml (hunk 3 above)
Comment (vladyslavyarets) [inline open-api/src/main/resources/yaml/openapi.yaml]: This open-api inplementation will be common for few modules, so you no need to specify reader module here.

**Review by vladyslavyarets — CHANGES_REQUESTED** [review]

**Review by vladyslavyarets — APPROVED** [review]

**Review by gendalf-fx — APPROVED** [review]

**Pull request merged**